scripts/daq_utils.py

```diff
@@ -0,0 +1,214 @@
+####################################################
+# Utilities to run the daq and view its status.
+#
+# Do not use non-standard python.
+####################################################
+import subprocess
+from subprocess import PIPE
+import socket
+import os, sys
+import getpass
+import time
+
+DAQBATCH_HUTCHES = ["tmo", "rix"]
+LOCALHOST = socket.gethostname()
+SLURM_PARTITION = "drpq"
+SLURM_JOBNAME = "submit_daq"
+DAQBATCH_SCRIPT = "submit_daq.sh"
+DAQBATCH_OUTPUT = "slurm_daq.log"
+MAX_RETRIES = 10
+
+
+def silentremove(filename):
+    try:
+        os.remove(filename)
+    except OSError as e:
+        if e.errno != errno.ENOENT:  # errno.ENOENT = no such file or directory
+            raise
+
+
+def call_subprocess(*args):
+    # FIXME: potentialy wanted to replace with 
+    # return subprocess.check_output(args, stderr=PIPE, universal_newlines=True)
+    # current block: return values are not in expected format. 
+    cc = subprocess.run(args, stdout=PIPE, stderr=PIPE)
+    output = None
+    if not cc.returncode:
+        output = str(cc.stdout.strip(), "utf-8")
+    return output
```

Comment:
If this can't be replaced with a version that raises on exception, the next best thing is to check for "None" wherever this is used. The important thing I was trying to get at was that there were places elsewhere that assume the result is a `str` and then the user could end up making really bad bug reports like:

`AttributeError: 'NoneType' object has no attribute 'splitlines'`

And then you or me needs to dig into this and figure out what the actual bug is. But if this raises a `CalledProcessError` instead, you get an error message from the script you're calling, which gives us a head start on debugging and can even help the user fix their own problems.

To keep the format the same I would suggest adding `check=True` as an argument to make this raise the helpful exception message.

Reply:
The output without automatic conversion works. This should untrap all the subprocess errors now.